repo/remote.go

```diff
@@ -319,7 +319,7 @@ func defaultHeadCmd(remote, vcs string) (string, error) {
 		cmd := exec.Command("git", "ls-remote", remote, "HEAD")
 		out, err := cmd.Output()
 		if err != nil {
-			return "", err
+			return "", fmt.Errorf("git ls-remote for %s : %v : %s", remote, err, err.(*exec.ExitError).Stderr)
```

Comment:
Huh, why did I think the error returned by `cmd.Output` included `Stderr` when it was formatted? I may need to fix some things elsewhere.

The type assertion makes me nervous. I don't think the error is necessarily `*exec.ExitError`. The doc says "usually".

Instead, lets just set `cmd.Stderr = os.Stderr`. We should print stderr anyway, even on success.

Reply:
I think printing out stderr for `go list -find -f {{.Module.Path}}` may be too much. It appears that go list prints all of its downloading information to stderr, which can become very noisy if we are generating multiple `go_repository` rules at once.
```
go: downloading github.com/jfrog/jfrog-client-go v0.3.3
go: finding github.com/emirpasic/gods v1.12.0
go: finding github.com/chzyer/logex v1.1.10
go: finding github.com/buger/jsonparser v0.0.0-20180910192245-6acdf747ae99
go: finding github.com/chzyer/test v0.0.0-20180213035817-a1ea475d72b1
go: finding github.com/spf13/viper v1.2.1
go: finding github.com/ulikunitz/xz v0.5.4
go: finding github.com/dsnet/compress v0.0.0-20171208185109-cc9eb1d7ad76
go: finding golang.org/x/net v0.0.0-20180926154720-4dfa2610cdf3
go: finding github.com/pierrec/lz4 v2.0.5+incompatible
go: finding github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
go: finding gopkg.in/src-d/go-billy.v4 v4.3.0
go: finding github.com/src-d/gcfg v1.3.0
go: finding golang.org/x/crypto v0.0.0-20181001203147-e3636079e1a4
go: finding gopkg.in/src-d/go-git.v4 v4.7.0
go: finding github.com/nwaples/rardecode v0.0.0-20171029023500-e06696f847ae
go: finding github.com/spf13/pflag v1.0.2
go: finding golang.org/x/sys v0.0.0-20180906133057-8cf3aee42992
go: finding github.com/mholt/archiver v2.1.0+incompatible
go: finding github.com/mitchellh/mapstructure v1.0.0
go: finding github.com/spf13/cast v1.2.0
go: downloading github.com/xanzy/ssh-agent v0.2.0
go: downloading golang.org/x/crypto v0.0.0-20181001203147-e3636079e1a4
go: downloading github.com/mholt/archiver v2.1.0+incompatible
go: downloading github.com/golang/snappy v0.0.0-20180518054509-2e65f85255db
go: downloading github.com/pierrec/lz4 v2.0.5+incompatible
go: downloading github.com/nwaples/rardecode v0.0.0-20171029023500-e06696f847ae
go: downloading github.com/dsnet/compress v0.0.0-20171208185109-cc9eb1d7ad76
go: downloading github.com/ulikunitz/xz v0.5.4
go: downloading gopkg.in/src-d/go-git.v4 v4.7.0
go: downloading github.com/src-d/gcfg v1.3.0
go: downloading github.com/emirpasic/gods v1.12.0
go: downloading gopkg.in/src-d/go-billy.v4 v4.3.0
go: downloading golang.org/x/net v0.0.0-20180926154720-4dfa2610cdf3
go: downloading github.com/spf13/viper v1.2.1
go: downloading github.com/mitchellh/mapstructure v1.0.0
go: downloading github.com/spf13/pflag v1.0.2
go: downloading github.com/spf13/cast v1.2.0
go: downloading gopkg.in/yaml.v2 v2.2.1
go: downloading golang.org/x/sys v0.0.0-20180906133057-8cf3aee42992
go: downloading github.com/buger/jsonparser v0.0.0-20180910192245-6acdf747ae99
go: finding github.com/codegangsta/cli v1.20.0
go: downloading github.com/codegangsta/cli v1.20.0
```

Maybe instead I can just prevent the type assertion from causing any panics.

Reply:
Oof. Yep, let's not print stderr on success and just fix the type assertion then.

This is golang/go#26152 for reference.